dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/internal/utils/MapUtils.kt

```diff
@@ -12,7 +12,7 @@ package com.datadog.android.core.internal.utils
  * Special value for missing attribute.
  */
 @Suppress("PLATFORM_CLASS_MAPPED_TO_KOTLIN")
-val NULL_MAP_VALUE: Object = Object()
+internal val NULL_MAP_VALUE: Object = Object()
```

Comment:
I think it is a blocking change. This one should be unique, because usage site in the core is `when (item) NULL_MAP_VALUE -> ...`, which is translated to `Intrinsics.areEqual(item, MapUtilsKt.getNULL_MAP_VALUE())`, which under the hood is doing `first == null ? second == null : first.equals(second)`, so objects are compared by reference.

In this PR `NULL_MAP_VALUE` is redeclared in Logs and Traces, and while for Trace it is fine, since the usage there doesn't go to the core, in case of Logs supplying another object will break the flow.